test/events.js

```diff
@@ -412,6 +428,141 @@ describe('the handleEvent() function', function () {
     });
   });
 
+  it('correctly processes events without a `ts` property when $BASKET_SQS_DISABLED_AFTER_TIMESTAMP is not set', () => {
+    mocks.mockBasketResponse({
+      reqheaders: { 'content-type': 'application/json' }
+    }).post('/fxa-activity/', function (body) {
+      assert.deepEqual(body, {
+        activity: 'account.login',
+        service: SERVICE,
+        fxa_id: UID,
+        first_device: true,
+        user_agent: USER_AGENT,
+        metrics_context: {}
+      });
+      return true;
+    }).reply(200, {
+      status: 'ok'
+    });
+    return events.handleEvent({
+      event: 'login',
+      service: SERVICE,
+      uid: UID,
+      email: EMAIL,
+      deviceCount: 1,
+      userAgent: USER_AGENT,
+      del: MAYBE_NEXT_TICK
+    });
+  });
+
+  it('correctly processes events without a `ts` property when $BASKET_SQS_DISABLED_AFTER_TIMESTAMP is set', () => {
+    mocks.mockBasketResponse({
+      reqheaders: { 'content-type': 'application/json' }
+    }).post('/fxa-activity/', function (body) {
+      assert.deepEqual(body, {
+        activity: 'account.login',
+        service: SERVICE,
+        fxa_id: UID,
+        first_device: true,
+        user_agent: USER_AGENT,
+        metrics_context: {}
+      });
+      return true;
+    }).reply(200, {
+      status: 'ok'
+    });
+    return utils.withEnviron({ BASKET_SQS_DISABLED_AFTER_TIMESTAMP: '2016-01-02 03:14:15' }, () => {
+      return utils.withFreshModules(require, ['../lib/events', '../lib/config'], () => {
+        const events = require('../lib/events');
+        return events.handleEvent({
+          event: 'login',
+          service: SERVICE,
+          uid: UID,
+          email: EMAIL,
+          deviceCount: 1,
+          userAgent: USER_AGENT,
+          del: MAYBE_NEXT_TICK
+        });
+      });
+    });
+  });
+
+  it('correctly processes events with `ts` less than $BASKET_SQS_DISABLED_AFTER_TIMESTAMP', () => {
+    mocks.mockBasketResponse({
+      reqheaders: { 'content-type': 'application/json' }
+    }).post('/fxa-activity/', function (body) {
+      assert.deepEqual(body, {
+        activity: 'account.login',
+        service: SERVICE,
+        fxa_id: UID,
+        first_device: true,
+        user_agent: USER_AGENT,
+        metrics_context: {}
+      });
+      return true;
+    }).reply(200, {
+      status: 'ok'
+    });
+    const cutover = '2016-01-02 03:14:15';
+    return utils.withEnviron({ BASKET_SQS_DISABLED_AFTER_TIMESTAMP: cutover }, () => {
+      return utils.withFreshModules(require, ['../lib/events', '../lib/config'], () => {
+        const events = require('../lib/events');
+        return events.handleEvent({
+          event: 'login',
+          ts: moment(cutover).subtract(2, 'minutes').valueOf() / 1000,
```

Comment:
Would the assertion here be more valuable if we pushed right up against the boundary and made this test minus 1 second rather than minus 2 minutes?